### ISSUE TITLE
How can I use custom layer in preprocessing_layers

### ISSUE THREAD

**Issue by HuPeng-Valer** [tensorflow/agents#515]
Hi, I want to add a custom resnet with attention in preprocessing_layers for encoding_net,but I got a valueError:Unknown layer:ResNet,this error occurred while executing function class_and_config_for_serialized_keras_object in file tensorflow.python.keras.utils.generic_utils.py  
here is the main code 

class ResNet(layers.Layer):

    def __init__(self, blocks, layer_dims=[2,2,2,2]):
        super(ResNet, self).__init__()
        self.in_channels = 16

        self.stem = Sequential([regularized_padded_conv(16, kernel_size=3, strides=1),
                                layers.BatchNormalization()])

        self.layer1 = self.build_resblock(blocks, 16,   layer_dims[0], stride=1)
        self.layer2 = self.build_resblock(blocks, 32,  layer_dims[1], stride=2)
        self.layer3 = self.build_resblock(blocks, 64,  layer_dims[2], stride=2)
        self.layer4 = self.build_resblock(blocks, 128,  layer_dims[3], stride=2)

    def build_resblock(self, blocks, out_channels, num_blocks, stride):
        strides = [stride] + [1] * (num_blocks - 1)          
        res_blocks = Sequential()

        for stride in strides:
            res_blocks.add(blocks(self.in_channels, out_channels, stride))
            self.in_channels = out_channels

        return res_blocks

    def get_config(self):
        config = super().get_config().copy()
        config.update({
            'in_channels':self.in_channels
        })
        return config
    

    def call(self,inputs, training=False):
        out = self.stem(inputs, training)
        out = tf.nn.relu(out)

        out = self.layer1(out, training=training)
        out = self.layer2(out, training=training)
        out = self.layer3(out, training=training)
        out = self.layer4(out, training=training)
        return out

here is the preprocessing_layer:

preprocessing_layers_attention = {
        'picture':
tf.keras.models.Sequential([ResNet(BasicBlock),
                                              tf.keras.layers.Flatten(),
                                              tf.keras.layers.Dropout(0.2),
                                              tf.keras.layers.Dense(256,activation=tf.keras.activations.relu)
                                            ]),
        'state':
tf.keras.layers.Dense(64)
    }

**Comment by kbanoop** [comment]
Can you try if using the sequential_layer from TF-Agents works?

Please see the tests here for an example:
https://github.com/tensorflow/agents/blob/master/tf_agents/networks/encoding_network_test.py

**Comment by sguada** [comment]
Reopen if needed.

**Comment by samarth-robo** [comment]
IMO this issue is not solved by using the TF-Agents Sequential network. It is caused by using a custom layer as a preprocessing combiner or preprocessing layer in the Actor or Critic. I also faced it. The solution that worked for me was to create actor and critic wrappers that included the custom layer:
```python
from tf_agents.networks.network import Network, DistributionNetwork
from tensorflow.keras.layers import Layer
import tensorflow as tf
from copy import deepcopy

class MyLayer(Layer):
# define your layer here

# actor using MyLayer
class MyActorNetwork(DistributionNetwork):
  def __init__(name='MyActorNetwork', my_arg, **adn_kwargs):
    input_spec = deepcopy(adn_kwargs['input_tensor_spec'])
    self.mylayer = MyLayer(my_arg)
    
    # output of mylayer is fed to the ActorDistributionNetwork, so ts should describe mylayer output
    # it will serve as input_tensor_spec for the ActorDistributionNetwork
    # ts = tf.TensorSpec(... <---- finish this
    adn_kwargs['input_tensor_spec'] = ts
    
    self.adn = ActorDistributionNetwork(name=f'{name}_actor', **adn_kwargs)

    # assume the layer is stateless and the entire state_spec describes the ActorDistributionNetwork's state
    super(MyActorNetwork, self).__init__(input_spec, self.adn.state_spec, self.adn.output_spec, name)

# critic using MyLayer
class MyCriticNetwork(Network):
## follow similar process as MyActorNetwork
```
